lib/connection.js

```diff
@@ -923,27 +952,34 @@ Connection.prototype.close = function(force, callback) {
   }
 
   return promiseOrCallback(callback, cb => {
-    this._close(force, cb);
+    this._close(force, false, cb);
   });
 };
 
 /**
  * Handles closing the connection
  *
  * @param {Boolean} force
+ * @param {Boolean} destroyConnAfterClose
  * @param {Function} callback
  * @api private
  */
-Connection.prototype._close = function(force, callback) {
+Connection.prototype._close = function(force, destroyConnAfterClose, callback) {
   const _this = this;
   const closeCalled = this._closeCalled;
   this._closeCalled = true;
+  this._destroyCalled = destroyConnAfterClose;
   if (this.client != null) {
     this.client._closeCalled = true;
+    this.client._destroyCalled = destroyConnAfterClose;
   }
 
+  const conn = this;
   switch (this.readyState) {
     case STATES.disconnected:
+      if (destroyConnAfterClose && this.base.connections.indexOf(conn) != -1) {
```

Comment:
```suggestion
      if (destroyConnAfterClose && this.base.connections.indexOf(conn) !== -1) {
```
indexOf always returns a number, so we can use tripple equal :)